Sources/MapboxDirections/RouteResponse.swift

```diff
@@ -107,8 +107,16 @@ extension RouteResponse: Codable {
             // The response lists the same number of tracepoints as the waypoints in the request, whether or not a given waypoint is leg-separating.
             waypoints = zip(decodedWaypoints, optionsWaypoints).map { (pair) -> Waypoint in
                 let (decodedWaypoint, waypointInOptions) = pair
-                let waypoint = Waypoint(coordinate: decodedWaypoint.coordinate, coordinateAccuracy: waypointInOptions.coordinateAccuracy, name: waypointInOptions.name?.nonEmptyString ?? decodedWaypoint.name)
+                let waypoint = Waypoint(coordinate: decodedWaypoint.coordinate,
+                                        coordinateAccuracy: waypointInOptions.coordinateAccuracy,
+                                        name: waypointInOptions.name?.nonEmptyString ?? decodedWaypoint.name)
+
+                waypoint.targetCoordinate = waypointInOptions.targetCoordinate
+                waypoint.heading = waypointInOptions.heading
+                waypoint.headingAccuracy = waypointInOptions.headingAccuracy
                 waypoint.separatesLegs = waypointInOptions.separatesLegs
+                waypoint.allowsArrivingOnOppositeSide = waypointInOptions.allowsArrivingOnOppositeSide
```

Comment:
Aside from the tests below, what would be a good way to prevent a future option from getting overlooked this way? How about copying the waypointInOptions and overwriting the fewer properties that are available in the decodedWaypoint. This could be good fodder for tail work; not essential for this PR.

Reply:
Then again, #388 would make a finer distinction between a requested waypoint and a received waypoint, such that the developer would be responsible for keeping track of the requested waypoints’ details separately, mirroring the API response format. But until that change lands someday, this is a good change to have for consistency.

Reply:
I was thinking about using a copy of `waypointInOptions` and assigning only minimum number of essential properties, but `public let coordinate: CLLocationCoordinate2D` prevents me from assigning it from `decodedWaypoint`. I'm not sure why coordinate is declared as `let` whereas others use `var`, but if that's okay to make coordinate var as well it'd make it possible.